ionoscloud/data_source_certificate_manager.go

```diff
@@ -0,0 +1,116 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	certmanager "github.com/ionos-cloud/sdk-cert-go"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cert"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCertificate() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCertificateRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"certificate": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"certificate_chain": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceCertificateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).CertManagerClient
+
+	var name, idStr string
+	id, idOk := d.GetOk("id")
+	if idOk {
+		idStr = id.(string)
+	}
+	t, nameOk := d.GetOk("name")
+	if nameOk {
+		name = t.(string)
+	}
+
+	var certificate certmanager.CertificateDto
+	var err error
+	var apiResponse *certmanager.APIResponse
+
+	if !idOk && !nameOk {
+		return diag.FromErr(fmt.Errorf("either id, or name must be set"))
+	}
+
+	if idOk {
+		certificate, apiResponse, err = client.GetCertificate(ctx, idStr)
+		certManagerLogApiResponse(apiResponse)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error getting certificate with id %s %w", idStr, err))
+		}
+		if nameOk {
+			if certificate.Properties != nil && certificate.Properties.Name != nil && *certificate.Properties.Name != name {
+				return diag.FromErr(fmt.Errorf("name of cert (UUID=%s, name=%s) does not match expected name: %s",
+					*certificate.Id, *certificate.Properties.Name, name))
+			}
+		}
+		if certificate.Properties != nil {
+			log.Printf("[INFO] Got certificate [Name=%s]", *certificate.Properties.Name)
+		}
+
+	} else {
+		certificates, apiResponse, err := client.ListCertificates(ctx)
+		certManagerLogApiResponse(apiResponse)
+
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occured while fetching certificates: %w ", err))
+		}
+
+		var results []certmanager.CertificateDto
+
+		if nameOk && certificates.Items != nil {
+			var certsFound []certmanager.CertificateDto
```

Comment:
don't think this is necessary since you don't have multiple search criteria and you can use only the results variable